app/assets/javascripts/components/autocomplete.js

```diff
@@ -73,13 +74,41 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         var options = [].filter.call($select.options, function (option) {
           return option.value === value
         })
+        var $previewContainer = $module.querySelector('.govuk-inset-text')
 
         if (options.length) {
           options[0].selected = true
-        }
 
-        if ($insertButton && contactSnippetTemplate && value) {
-          $insertButton.dataset.modalData = contactSnippetTemplate.replace('#', value)
+          // Prepare snippet for insertion
+          if ($insertButton && contactSnippetTemplate && value) {
+            var contactSnippet = contactSnippetTemplate.replace('#', value)
+            $insertButton.dataset.modalData = contactSnippet
+          }
+
+          // Generate/update contact preview
+          var govspeakPath = document.querySelector('[data-govspeak-path]').dataset.govspeakPath
+          if (contactSnippet && govspeakPath && window.FetchContent) {
```

Comment:
Again, wondering why we're checking the helper module exists. Could that not be the case?